src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -265,13 +263,20 @@ private static void createImpedantBranch(LfBranch branch, LfBus bus1, LfBus bus2
 
     private static void createBranchEquations(LfNetwork network, VariableSet variableSet, AcEquationSystemCreationParameters creationParameters,
                                               EquationSystem equationSystem) {
+        // memorize already created non impedant branches to avoid creating parallel non impedant
+        Set<String> nonImpedantBranchesAlreadyCreated = new HashSet<>();
+
         for (LfBranch branch : network.getBranches()) {
             LfBus bus1 = branch.getBus1();
             LfBus bus2 = branch.getBus2();
             PiModel piModel = branch.getPiModel();
             if (piModel.getZ() < DcEquationSystem.LOW_IMPEDANCE_THRESHOLD) {
                 if (bus1 != null && bus2 != null) {
-                    createNonImpedantBranch(variableSet, equationSystem, branch, bus1, bus2);
+                    String bus1Bus2Key = bus1.getNum() < bus2.getNum() ? bus1.getNum() + "_" + bus2.getNum() : bus2.getNum() + "_" + bus1.getNum();
+                    if (!nonImpedantBranchesAlreadyCreated.contains(bus1Bus2Key)) {
```

Comment:
I am not sure to well understand what does it mean: 
- When you want to create a low impedance branch you check if a similar branch already exists between the same bus ;
- If yes, with the same rho and angle, what do you do ? Nothing ? We delete the branch ?

Reply:
This is just at equation level, so we do not change the network for avoid to create redundant equation. I am going to investigate more deeply why we need to skip redundant coupling equation to better understand what the problem is.
Also could you commit your test case with non impedant branch loop?